mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -334,73 +325,53 @@ func resourceMongoDBAtlasSearchIndexRead(ctx context.Context, d *schema.Resource
 	return nil
 }
 
-func flattenSearchIndexSynonyms(synonyms []map[string]interface{}) []map[string]interface{} {
-	synonymsMap := make([]map[string]interface{}, 0)
-
-	for _, s := range synonyms {
-		sourceCollection := s["source"].(map[string]interface{})
-		synonym := map[string]interface{}{
-			"name":              s["name"],
-			"analyzer":          s["analyzer"],
-			"source_collection": sourceCollection["collection"],
+func flattenSearchIndexSynonyms(synonyms []admin.SearchSynonymMappingDefinition) []map[string]any {
+	synonymsMap := make([]map[string]any, len(synonyms))
+	for i, s := range synonyms {
+		synonymsMap[i] = map[string]any{
+			"name":              s.Name,
+			"analyzer":          s.Analyzer,
+			"source_collection": s.Source.Collection,
 		}
-		synonymsMap = append(synonymsMap, synonym)
 	}
 	return synonymsMap
 }
 
-func marshallSearchIndexAnalyzers(fields []map[string]interface{}) (string, error) {
-	if len(fields) == 0 {
-		return "", nil
-	}
-
-	mappingFieldJSON, err := json.Marshal(fields)
-	return string(mappingFieldJSON), err
-}
-
-func marshallSearchIndexMappingsField(fields map[string]interface{}) (string, error) {
-	if len(fields) == 0 {
-		return "", nil
-	}
-
-	mappingFieldJSON, err := json.Marshal(fields)
-	return string(mappingFieldJSON), err
+func marshalSearchIndex(fields any) (string, error) {
```

Comment:
why are we changing from `[]map[string]interface{}` to `any`?

Reply:
maybe better `[]map[string]any`?

Reply:
we could rename the function directly to something more representative, like `toJsonString`

Reply:
I like the idea of calling Marshal as there is an opossed method to unmarshal,   and SearchIndex is meant to be used in this resource.  so marshal as packaging the info (that turns out to be in json) and unmarshaling to unpackage it